File: xssfinder-java-executor/src/test/java/org/xssfinder/scanner/ThriftToReflectionLookupTest.java
package org.xssfinder.scanner;

import org.junit.Test;

import java.lang.reflect.Method;

import static org.hamcrest.Matchers.is;
import static org.hamcrest.Matchers.nullValue;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertThat;

public class ThriftToReflectionLookupTest {
    private static final String PAGE_ID = "SomePage";
    private static final String METHOD_ID = "someMethod";
    @Test
    public void pageMappingIsEmptyOnConstruction() {
        // given
        ThriftToReflectionLookup lookup = new ThriftToReflectionLookup();

        // when
        Class<?> pageClass = lookup.getPageClass(PAGE_ID);

        // then
        assertThat(pageClass, is(nullValue()));
    }

    @Test
    public void methodMappingIsEmptyOnConstruction() {
        // given
        ThriftToReflectionLookup lookup = new ThriftToReflectionLookup();

        // when
        Method method = lookup.getMethod(METHOD_ID);

        // then
        assertThat(method, is(nullValue()));
    }

    @Test
    public void setPageClassIsAvailableFromLookup() {
        // given
        ThriftToReflectionLookup lookup = new ThriftToReflectionLookup();
        lookup.putPageClass(PAGE_ID, SomePage.class);

        // when
        Class<?> pageClass = lookup.getPageClass(PAGE_ID);

        // then
        assertEquals(pageClass, SomePage.class);
    }

    @Test
    public void setMethodIsAvailableFromLookup() throws Exception {
        // given
        ThriftToReflectionLookup lookup = new ThriftToReflectionLookup();
        Method goToSomePage = SomePage.class.getMethod("goToSomePage");
        lookup.putMethod(METHOD_ID, goToSomePage);

        // when
        Method method = lookup.getMethod(METHOD_ID);

        // then
        assertEquals(method, goToSomePage);
    }

    private static class SomePage {
        public SomePage goToSomePage() { return null; }
    }
}

File: xssfinder-php-executor/test/XssFinder/Scanner/ThriftToReflectionLookupTest.php
<?php

namespace XssFinder\Scanner;

use PHPUnit_Framework_TestCase;
use ReflectionClass;

class ThriftToReflectionLookupTest extends PHPUnit_Framework_TestCase
{
    const PAGE_ID = "SomePage";
    const METHOD_ID = "someMethod";

    function testPageMappingIsEmptyOnConstruction()
    {
        // given
        $lookup = new ThriftToReflectionLookup();

        // when
        $pageClass = $lookup->getPageClass(self::PAGE_ID);

        // then
        assertThat($pageClass, is(nullValue()));
    }

    function testMethodMappingIsEmptyOnConstruction()
    {
        // given
        $lookup = new ThriftToReflectionLookup();

        // when
        $method = $lookup->getMethod(self::METHOD_ID);

        // then
        assertThat($method, is(nullValue()));
    }

    function testSetPageClassIsAvailableFromLookup()
    {
        // given
        $lookup = new ThriftToReflectionLookup();
        $reflectionClass = new ReflectionClass('\XssFinder\Scanner\TTRL_TestPages_SomePage');
        $lookup->putPageClass(self::PAGE_ID, $reflectionClass);

        // when
        $pageClass = $lookup->getPageClass(self::PAGE_ID);

        // then
        assertThat($pageClass, is($reflectionClass));
    }

    function testSetMethodIsAvailableFromLookup()
    {
        // given
        $lookup = new ThriftToReflectionLookup();
        $reflectionClass = new ReflectionClass('\XssFinder\Scanner\TTRL_TestPages_SomePage');
        $someMethod = $reflectionClass->getMethod('someMethod');
        $lookup->putMethod(self::METHOD_ID, $someMethod);

        // when
        $method = $lookup->getMethod(self::METHOD_ID);

        // then
        assertThat($method, is($someMethod));
    }
}

class TTRL_TestPages_SomePage {
    function someMethod() {}
}